### DISCUSSION TITLE
What does the 1 in nn.Parameter(torch.randn(1, requires_grad=True, dtype=torch.float) mean

### DISCUSSION THREAD

**Discussion by D1Asif** [mrdbourke/pytorch-deep-learning#578] [Q&A]
```python
import torch
from torch import nn


class LinearRegressionModel(nn.Module): 
  def __init__(self):
    super().__init__()
    self.weights = nn.Parameter(torch.randn(1,
                                            requires_grad=True,
                                            dtype = torch.float))
    self.bias = nn.Parameter(torch.randn(1,
                                         requires_grad=True,
                                         dtype=torch.float))
  def forward(self, x: torch.Tensor) -> torch.Tensor: 
    return self.weights * x + self.bias```

**Comment by Ghani-Haider** [comment]
It is the size parameter. As per torch's documentation on [torch.randn](https://pytorch.org/docs/stable/generated/torch.randn.html):

size (int...) – a sequence of integers defining the shape of the output tensor. Can be a variable number of arguments or a collection like a list or tuple.

**Reply by D1Asif** [reply]
Yeah I've seen that the documentation but don't understand what it means

**Reply by wittyalias** [reply]
This is pretty well explained in the 00_pytorch_fundamentals.ipynb. Check out the text under code block number 13. 
But in brief, the size parameter gives both the number of dimensions in a tensor as well as the number of variables in it. An easy way to think about the dimensions is as rows and columns. 
- torch.rand(1) has no rows or columns, its just a single number, so it's a 0-dimensional tensor, with the one number in it - that is, a scalar. 
- torch.rand(4) has 4 numbers in a row, so it's a 1-dimensional tensor, with four numbers, so, a vector of length 4. 
- torch.rand((2,3) shows that there are two rows and three columns, so that's a 2-dimensional tensor, with a total of 6 numbers in it.
